pennylane/templates/state_preparations/mottonen.py

```diff
@@ -128,102 +134,76 @@ def _uniform_rotation_dagger(gate, alpha, control_wires, target_wire):
-    """
-
-    _uniform_rotation_dagger(qml.RZ, alpha, control_wires, target_wire)
-
-
-def _uniform_rotation_y_dagger(alpha, control_wires, target_wire):
-    """Applies the inverse of a Y rotation to the target qubit
-    that is uniformly controlled by the control qubits.
-
-    Args:
-        alpha (array[float]): alpha parameters
-        control_wires (array[int]): wires that act as control
-        target_wire (int): wire that acts as target
-    """
-
-    _uniform_rotation_dagger(qml.RY, alpha, control_wires, target_wire)
-
-
 def _get_alpha_z(omega, n, k):
-    r"""Computes the rotation angles alpha for the Z rotations.
+    r"""Computes the rotation angles for the Z rotations applied to the k'th qubit.
+
+    .. math:: \alpha^{z,k}_j = \sum_{l=1}^{2^{k-1}} \frac{\omega_{(2j-1) 2^{k-1}+l} - \omega_{(2j-2) 2^{k-1}+l}}{2^{k-1}}
 
     Args:
-        omega (float): phase of the input
+        omega (float): phases of the input
         n (int): total number of qubits
         k (int): index of current qubit
 
     Returns:
-        scipy.sparse.dok_matrix[np.float64]: a sparse vector representing :math:`\alpha^z_k`
+        array representing :math:`\alpha^{z,k}`
     """
-    alpha_z_k = sparse.dok_matrix((2 ** (n - k), 1), dtype=np.float64)
+    alpha_z_k = np.zeros((2 ** (n - k),), dtype=np.float64)
 
-    for (i, _), om in omega.items():
-        i += 1
-        j = int(np.ceil(i * 2 ** (-k)))
+    for i in range(len(omega)):
+        j = int(np.ceil((i + 1) * 2 ** (-k)))
         s_condition = 2 ** (k - 1) * (2 * j - 1)
-        s_i = 1.0 if i > s_condition else -1.0
-        alpha_z_k[j - 1, 0] = alpha_z_k[j - 1, 0] + s_i * om / 2 ** (k - 1)
+        s_i = 1.0 if (i + 1) > s_condition else -1.0
+        alpha_z_k[j - 1] = alpha_z_k[j - 1] + s_i * omega[i] / 2 ** (k - 1)
 
     return alpha_z_k
 
 
 def _get_alpha_y(a, n, k):
-    r"""Computes the rotation angles alpha for the Y rotations.
+    r"""Computes the rotation angles for the Y rotations applied to the k'th qubit.
+
+    The angles are related to a as:
+
+    .. math:: \alpha^{y,k}_j = 2 \arcsin \sqrt{ \frac{ \sum_{l=1}^{2^{k-1}} a_{(2j-1)2^{k-1} +l}^2  }{ \sum_{l=1}^{2^{k}} a_{(j-1)2^{k} +l}^2  } }
 
     Args:
-        omega (float): phase of the input
+        a (float): absolute values of the input
         n (int): total number of qubits
         k (int): index of current qubit
 
     Returns:
-        scipy.sparse.dok_matrix[np.float64]: a sparse vector representing :math:`\alpha^y_k`
+        array representing :math:`\alpha^{y,k}`
     """
 
-    alpha = sparse.dok_matrix((2 ** (n - k), 1), dtype=np.float64)
-
-    numerator = sparse.dok_matrix((2 ** (n - k), 1), dtype=np.float64)
-    denominator = sparse.dok_matrix((2 ** (n - k), 1), dtype=np.float64)
+    numerator = np.zeros((2 ** (n - k),), dtype=np.float64)
+    denominator = np.zeros((2 ** (n - k),), dtype=np.float64)
+    alpha = np.zeros((2 ** (n - k),), dtype=np.float64)
```

Comment:
```suggestion
    numerator = np.zeros(2 ** (n - k), dtype=np.float64)
    denominator = np.zeros(2 ** (n - k), dtype=np.float64)
    alpha = np.zeros(2 ** (n - k), dtype=np.float64)
```
?

Reply:
The reason to pass a tuple here is that it generalises better to the use of a classical wrapper function. But we can commit your suggestion and change it to take the shape in following PRs.